### PR TITLE
refactor(transformer/typescript): insert assignments after super by `StatementInjector`

### PR DESCRIPTION
**Pull request by Dunqing**

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6654?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6654** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6654?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6653** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6653?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6652** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6652?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by overlookmotel** [comment]
bench
Nice!

**Comment by Dunqing** [comment]
> bench Nice!

Why I can't see the performance improvement in this PR

File: crates/oxc_transformer/src/typescript/annotations.rs [hunk 1]
```diff
@@ -417,29 +439,6 @@ impl<'a, 'ctx> Traverse<'a> for TypeScriptAnnotations<'a, 'ctx> {
             // Ignore ModuleDeclaration as it's handled in the program
             _ => true,
         });
```
Comment (Dunqing) [inline crates/oxc_transformer/src/typescript/annotations.rs:441]: Do you think we to have a more general `Statement` helper to cover this usage?  We can mark a statement as `Delete` in `exit_statement`, this way we don't need to call `retain` each `exit_statements` @overlookmotel cc?

Comment (Dunqing) [reply crates/oxc_transformer/src/typescript/annotations.rs:441]: Oh, you may need to open the file to look at the code

**Comment by Dunqing** [comment]
Just a tiny refactor, so I am going to merge it without approvement

**Comment by Dunqing** [comment]
### Merge activity

* **Oct 24, 1:13 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**